integration/test/basic/basic_test.go

```diff
@@ -27,39 +32,55 @@ func TestChartInstalled(t *testing.T) {
 		RestConfig:      f.RestConfig(),
 		TillerNamespace: "giantswarm",
 	}
-	helmClient, err := helmclient.New(c)
+
+	gsHelmClient, err := helmclient.New(c)
 	if err != nil {
 		t.Fatalf("could not create helmClient %v", err)
 	}
 
-	err = f.InstallResource("chart-operator-resource", templates.ChartOperatorResourceValues, ":stable")
+	// Test Creation
+	log.Printf("creating %q", cr)
+	err = f.InstallResource(cr, templates.ChartOperatorResourceValues, ":stable")
+	if err != nil {
+		t.Fatalf("could not install %q %v", cr, err)
+	}
+
+	err = waitForReleaseStatus(gsHelmClient, testRelease, "DEPLOYED")
+	if err != nil {
+		t.Fatalf("could not get release status of %q %v", testRelease, err)
+	}
+	log.Printf("%q succesfully deployed", testRelease)
+
+	// Test Deletion
+	log.Printf("deleting %q", cr)
+	err = helmClient.DeleteRelease(cr)
 	if err != nil {
-		t.Fatalf("could not install chart-operator-resource-chart %v", err)
+		t.Fatalf("could not delete %q %v", cr, err)
 	}
 
-	var rc *helmclient.ReleaseContent
+	err = waitForReleaseStatus(gsHelmClient, testRelease, "DELETED")
+	if !helmclient.IsReleaseNotFound(err) {
+		t.Fatalf("%q not succesfully deleted %v", testRelease, err)
```

Comment:
Since we purge the release, we are actually not waiting for a "DELETED" Status but for a "release not found" error. Deletion takes a while, so we need to wait till the `waitForReleaseStatus`/backoff returns that error. Any ideas how to match that error faster are welcome.

Reply:
You could write a new functions which checks for the specific error in each backoff. 
This here should be fine though.

Reply:
Yes thought about it too, but felt too much like duplicate code, just to fail one layer earlier. Think I am going with this.